### PR TITLE
Updating tools/clair3 from version 0.1.11 to 0.1.12

### PR DESCRIPTION
**Pull request by planemo-autoupdate**
Hello! This is an automated update of the following tool: **tools/clair3**. I created this PR because I think the tool's main dependency is out of date, i.e. there is a newer version available through conda.

I have updated tools/clair3 from version 0.1.11 to 0.1.12.

**Project home page:** https://github.com/HKU-BAL/Clair3/releases

For any comments, queries or criticism about the bot, not related to the tool being updated in this PR, please create an issue [here](https://github.com/planemo-autoupdate/autoupdate/issues/new).

### CODE REVIEW

**Comment by wm75** [comment]
With https://github.com/HKU-BAL/Clair3/commit/4593d51cb411fedcbe36dae13acd8de0c925c0c3 support for cram input files got added to the command line tool.
So we could allow cram on top of bam as input format in the wrapper, too, if we create an appropriate .crai index symlink in that case, and a test would be good too, of course.

@gallardoalba @mira-miracoli or anyone else interested to add to this to the PR here?

**Comment by gallardoalba** [comment]
> With [HKU-BAL/Clair3@4593d51](https://github.com/HKU-BAL/Clair3/commit/4593d51cb411fedcbe36dae13acd8de0c925c0c3) support for cram input files got added to the command line tool. So we could allow cram on top of bam as input format in the wrapper, too, if we create an appropriate .crai index symlink in that case, and a test would be good too, of course.
> 
> @gallardoalba @mira-miracoli or anyone else interested to add this to the PR here?

Yes, I'll work on it.

File: tools/clair3/clair3.xml [hunk 1]
```diff
@@ -88,81 +97,75 @@
...
         </data>
-        <data name="full_alignment" format="vcf_bgzip" from_work_dir="./full_alignment.vcf.gz">
-            <filter>output_options['selection_mode'] == 'advanced' and output_options['full_alignment_check']</filter>
+        <data name="full_alignment" format="vcf_bgzip" from_work_dir="./full_alignment.vcf.gz" label="${tool.name} on ${on_string}: full alignment">
+            <filter>output_files and 'phased_bam' in output_files</filter>
         </data>
-        <data name="phased_bam" format="bam" from_work_dir="./phased_bam.bam">
-            <filter>output_options['selection_mode'] == 'advanced' and output_options['phased_bam_check']</filter>
+        <data name="phased_bam" format="bam" from_work_dir="./phased_bam.bam" label="${tool.name} on ${on_string}: phased BAM">
+            <filter>output_files and 'gvcf' in output_files</filter>
```
Comment (wm75) [inline tools/clair3/clair3.xml:168]: Is there something wrong with the filtering? Or why is "full_alignment" dependent on "phased_bam" and "phased_bam" on "gvcf"?

Comment (gallardoalba) [reply tools/clair3/clair3.xml:168]: It was my mistake sorry.

File: tools/clair3/clair3.xml [hunk 2]
```diff
@@ -43,16 +51,16 @@
...
                   --bed_fn=input.bed
             #elif $bed_or_vcf.bed_or_vcf_selector == 'vcf'
                   --vcf_fn=input.vcf
             #else
                 $bed_or_vcf.include_all_ctgs
             #end if
-            #if $output_options.selection_mode == 'advanced'
-                $output_options.gvcf
+            #if 'gvcf' in $output_files
+                --gvcf
```
Comment (wm75) [inline tools/clair3/clair3.xml:63]: This won't work now that you've moved the option down below.

File: tools/clair3/clair3.xml [hunk 3]
```diff
@@ -1,8 +1,12 @@
 <tool id="clair3" name="Clair3" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="21.09">
+    <description> germline small variant caller for long-reads</description>
```
Comment (wm75) [inline tools/clair3/clair3.xml:2]:
```suggestion
    <description>germline small variant caller for long-reads</description>
```

File: tools/clair3/clair3.xml [hunk 4]
```diff
@@ -88,81 +97,73 @@
...
                     <option value="r941_prom_hac_g360+g422">r941_prom_hac_g360+g422</option>
                     <option value="hifi">hifi</option>
                     <option value="ilmn">ilmn</option>
                 </param>
             </when>
             <when value="datatable">
-                <param argument="--model_path" type="select" name="model" label="Select model" >
+                <param  name="model" argument="--model_path" type="select" label="Model from database" >
                     <options from_data_table="clair3_models">
                         <validator type="no_options" message="no modles available" />
```
Comment (wm75) [inline tools/clair3/clair3.xml:115]:
```suggestion
                        <validator type="no_options" message="no cached models available" />
```

File: tools/clair3/clair3.xml [hunk 5]
```diff
@@ -88,81 +97,73 @@
...
+                <param type="select" name="select_built_in" label="Built-in model">
                     <option value="r941_prom_sup_g5014">r941_prom_sup_g5014</option>
                     <option value="r941_prom_hac_g360+g422">r941_prom_hac_g360+g422</option>
                     <option value="hifi">hifi</option>
                     <option value="ilmn">ilmn</option>
                 </param>
             </when>
             <when value="datatable">
-                <param argument="--model_path" type="select" name="model" label="Select model" >
+                <param  name="model" argument="--model_path" type="select" label="Model from database" >
```
Comment (wm75) [inline tools/clair3/clair3.xml:113]:
```suggestion
                <param  name="model" argument="--model_path" type="select" label="Cached model from server" >
```

File: tools/clair3/clair3.xml [hunk 6]
```diff
@@ -88,81 +97,73 @@
             </when>
         </conditional>
         <conditional name="model_source">
-            <param type="select" label="Select if you want to use built-in models or your own via datatable." name="source">
+            <param name="source" type="select" label="Clair3 model" help="Select if you want to use built-in models or your own via datatable.">
                 <option value="built-in">Built-in</option>
                 <option value="datatable">Datatable</option>
```
Comment (wm75) [inline tools/clair3/clair3.xml:102]:
```suggestion
            <param name="source" type="select" label="Clair3 model" help="Select if you want to use a built-in model shipping with the tool or a model cached on this Galaxy server.">
                <option value="built-in">Built-in</option>
                <option value="datatable">Cached</option>
```

File: tools/clair3/clair3.xml [hunk 7]
```diff
@@ -88,81 +97,73 @@
...
-            <param type="boolean" name="print_ref_calls" truevalue="--print_ref_calls" falsevalue="" label="Show reference calls (0/0) in vcf file, default: disable." />
-            <param type="select" name="ploidity_model" label="Call with the following ploidy model" help="EXPERIMENTAL: Enable haploid calling mode. Only 1/1 is considered as a variant, default: disable. EXPERIMENTAL: Enable haploid calling mode. 0/1 and 1/1 are considered as a variant, default: disable.">
-                <option value="" selected="true">diploid</option>
-                <option value="--haploid_sensitive">haploid (sensitive)</option>
-                <option value="--haploid_precise">haploid (precise)</option>
+        <param type="boolean" argument="--gvcf" truevalue="--gvcf" falsevalue="" label="Enable GVCF output" help="GVCF stands for Genomic VCF. A GVCF is a kind of VCF, so the basic format specification is the same as for a regular VCF but contains extra information. Default: disable" />
+        <param name="output_files" type="select" display="checkboxes" multiple="true" label="Output files">
```
Comment (wm75) [inline tools/clair3/clair3.xml:137]:
```suggestion
        <param name="output_files" type="select" display="checkboxes" multiple="true" label="Optional additional output files">
```

**Review by wm75 — COMMENTED** [review]
Just a few more spelling and wording fixes/suggestions, plus the issue with gvcf output.
But this looks *much* better already than the old version.

File: tools/clair3/clair3.xml (hunk 2 above)
Comment (gallardoalba) [reply tools/clair3/clair3.xml]: True, thanks.

**Review by wm75 — APPROVED** [review]
Perfect, thanks a lot!

**Pull request merged**